EmailMessageTrigger.trigger

```diff
@@ -0,0 +1,25 @@
+trigger EmailMessageTrigger on EmailMessage (before insert) {
+	//get the list of incoming messages
+    List<EmailMessage> incomingMessages = Trigger.new;
+    System.debug('Size of incoming messages ' + incomingMessages.size());
+    System.debug('Messaging ' + incomingMessages);
+ 	//process messages
+    for(EmailMessage message: incomingMessages){
+        if(message.ParentId==null) {continue;}
+     //get the list of possible parent cases
+        List<Case> parents = [
```

Comment:
@jivan0608 
here you have SOQL query inside a FOR LOOP.
This is a bad practice. Please use One soql query and Map collection functionality to proceed